recipes/libbigwig/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import collect_libs, get, copy
+from conan.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.50.0"
+
+
+class LibBigWigConan(ConanFile):
+    name = "libbigwig"
+    description = "A C library for handling bigWig files"
+    topics = ("bioinformatics", "bigwig", "bigbed")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/dpryan79/libBigWig"
+    license = "MIT"
+    settings = "arch", "build_type", "compiler", "os"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_curl": [True, False],
+        "with_zlibng": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_curl": False,
+        "with_zlibng": False
+    }
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+        # Workaround weird gsasl link error
+        # See: https://github.com/conan-io/conan-center-index/pull/12195#issuecomment-1214193211
+        if self.options.with_curl:
+            self.options["libcurl"].with_imap = False
+            self.options["libcurl"].with_pop3 = False
+            self.options["libcurl"].with_smtp = False
```

Comment:
```suggestion
```
Move it to validate instead:

```python
def validate(self):
    if self.options.with_curl:
        libcurl = self.dependencies["libcurl"]
        if libcurl.options.with_imap or libcurl.options.with_pop3 or libcurl.options.with_smtp:
            raise ConanInvalidConfiguration(f"{self.ref} requires libcurl using these follow options: -o libcurl:with_imap=False -o libcurl:with_pop3=False -o libcurl:with_smtp=False")
```

Try to be more explicit than implicit, Conan follows Python karma. Also, when failing form validate, instead of missing package ID when building, is much more explicit what is missing than trying to find which package property does not fit those available in the server.